pyccel/compilers/default_compilers.py

```diff
@@ -185,26 +185,42 @@ def change_to_lib_flag(lib):
     else:
         return lib
 
-python_version = sysconfig.get_python_version()
 config_vars = sysconfig.get_config_vars()
-linker_flags = [change_to_lib_flag(l) for l in
-                    config_vars.get("LIBRARY","").split() + \
-                    config_vars.get("LDSHARED","").split()[1:]]
+
 python_info = {
         "libs" : config_vars.get("LIBM","").split(), # Strip -l from beginning
-        "libdirs" : config_vars.get("LIBDIR","").split(),
         'python': {
             'flags' : config_vars.get("CFLAGS","").split()\
                 + config_vars.get("CC","").split()[1:],
             'includes' : [*config_vars.get("INCLUDEPY","").split(), get_numpy_include()],
-            'libs' : [l[2:] for l in linker_flags if l.startswith('-l')],
-            'libdirs' : [l[2:] for l in linker_flags if l.startswith('-L')]+config_vars.get("LIBPL","").split(),
-            "shared_suffix" : config_vars.get("EXT_SUFFIX",".so"),
+            "shared_suffix" : config_vars['EXT_SUFFIX'],
             }
         }
+
 if sys.platform == "win32":
-    python_info['python']['libs'].append('python{}'.format(config_vars["VERSION"]))
-    python_info['python']['libdirs'].extend(config_vars.get("installed_base","").split())
+    python_lib = os.path.join(config_vars["prefix"], 'python{}.dll'.format(config_vars["VERSION"]))
+    if os.path.exists(python_lib):
+        python_info['python']['dependencies'] = (python_lib,)
+    else:
+        python_info['python']['libs'] = ('python{}'.format(config_vars["VERSION"]),)
+        python_info['python']['libdirs'] = config_vars.get("installed_base","").split()
+else:
+    python_lib_base = os.path.join(config_vars["prefix"], "lib", config_vars["LDLIBRARY"])
+    possible_shared_lib = python_lib_base.replace('.a','.so')
+    possible_shared_lib = possible_shared_lib if os.path.exists(possible_shared_lib) else ''
+    possible_static_lib = python_lib_base.replace('.so','.a')
+    possible_static_lib = possible_static_lib if os.path.exists(possible_static_lib) else ''
+    if possible_shared_lib == '' and possible_static_lib == '':
+        linker_flags = [change_to_lib_flag(l) for l in
+                        config_vars.get("LIBRARY","").split() + \
+                        config_vars.get("LDSHARED","").split()[1:]]
+        python_info['python']['libs'] = [l[2:] for l in linker_flags if l.startswith('-l')]
+        python_info['python']['libdirs'] = [l[2:] for l in linker_flags if l.startswith('-L')] + \
+                            config_vars.get("LIBPL","").split()+config_vars.get("LIBDIR","").split()
+    elif possible_static_lib != '':
+        python_info['python']['dependencies'] = (possible_static_lib,)
+    else:
+        python_info['python']['dependencies'] = (possible_shared_lib,)
 
```

Comment:
Can you please explain what you are doing here?

Reply:
On some systems (notably with anaconda), the library mentioned in the configs does not exist but the equivalent static library does. This section of code searches for any libraries which have the expected name any actually exists. The static library is preferred as it means we don't need a lib dir argument which stops anaconda causing problems (eg linking scipy's blas library or h5py's hdf5 library). I have added comments: [a4139c6](https://github.com/pyccel/pyccel/pull/1107/commits/a4139c6a3b01bddd720cea70dccf45626819163c)

Reply:
Thanks!